fitbenchmarking/fitbenchmarking
# Issue #266: Enable user functions in Mantid problems

#### Description of Work

Fixes #266 

Enables user defined formulas in mantid.


#### Testing Instructions

1. Create a problem file with a UserFunction (`name=UserFunction,Formula=<some_Formula>,...`) then run `from fitbenchmarking.parsing.parser_factory import parse_problem_file` and create a fitting problem from the test problem.
   Check that eval_f behaves as you would expect.

Function: Does the change do what it's supposed to?
Tests: Does it pass? Is there adequate coverage for new code?
Style: Is the coding style consistent? Is anything overly confusing?
Documentation: Is there a suitable change to documentation for this change?
